src/events.js

```diff
@@ -1925,13 +1925,49 @@ export const addEventSpecimenLinkForm = (formData) => {
 
     if (document.getElementById('navBarParticipantCheckIn')) document.getElementById('navBarParticipantCheckIn').dataset.connectId = connectId;
 
-    form.addEventListener('submit', e => {
+    form.addEventListener('submit', async (e) => {
         e.preventDefault();
-        btnsClicked(connectId, formData);
+        const collections = await getCollectionsByVisitType(connectId, '266600170');
```

Comment:
Would prefer this to be modular instead of hardcoding baseline collection CID

Reply:
Resolved
https://github.com/episphere/biospecimen/pull/294/commits/43dd3db732689c038bf4ada84db6402063437c5b